src/main/java/org/testeditor/fixture/host/net/Connection.java

```diff
@@ -145,28 +181,12 @@ public void run() {
         }).start();
     }
 
-    /**
-     *
-     * @return true if connection is established to mainframe through
-     *         s3270/ws3270 emulation and process is still running, false
-     *         otherwise.
-     */
-    public boolean isConnected() {
-        if (s3270Process == null || in == null || out == null) {
-            return false;
-        } else {
-            final Result r = doCommand("");
-            if (r.getStatus().matches(". . . C.*")) {
-                return true;
-            } else {
-                doQuit();
-                s3270Process.destroy();
-                s3270Process = null;
-                in = null;
-                out = null;
-                return false;
-            }
-        }
+    private Status createStatus(Result r) {
```

Comment:
why not put this method into the class Result and internally set the status if not yet parsed from the status string?